### PR TITLE
test: fix test_close_accept flakiness on Centos5

### PR DESCRIPTION
**Pull request by santigimeno**
It is not guaranteed which stream will read the data first.

This test is failing in `Jenkins` with this error:

```
`tcp_close_accept` failed: exit code 6
Output from process `tcp_close_accept`:
Assertion failed in test/test-tcp-close-accept.c on line 88: stream == (uv_stream_t*) &tcp_incoming[0]
```

### CODE REVIEW

**Comment by bnoordhuis** [comment]
LGTM.  CI: https://ci.nodejs.org/job/libuv-test-commit/15/

**Comment by santigimeno** [comment]
Failing on a different place now: in a call to `uv_close` :(. I'll try to add some debug logs to find out what's going on.

**Comment by santigimeno** [comment]
PR updated with a fix

File: test/test-tcp-close-accept.c [hunk 1]
```diff
@@ -84,17 +84,17 @@ static void read_cb(uv_stream_t* stream, ssize_t nread, const uv_buf_t* buf) {
...
   ASSERT(1 == nread);
 
   loop = stream->loop;
   read_cb_called++;
 
-  /* Close all active incomings, except current one */
-  for (i = 1; i < got_connections; i++)
-    uv_close((uv_handle_t*) &tcp_incoming[i], close_cb);
+  /* Close the other active connection, not current one */
+  i = (stream == (uv_stream_t*) &tcp_incoming[0]) ? 1 : 0;
```
Comment (bnoordhuis) [inline test/test-tcp-close-accept.c]: You don't need the ternary, `x == y` always evaluates to either 0 or 1.  You could also write it as:
```c
i = (uv_tcp_t*) stream - &tcp_incoming[0];
```
That will keep on working when more entries are added to the `tcp_incoming` array.  You could do a similar thing for the ASSERT a few lines up.  Anyway, I'll leave it up to you if you want to change it.

**Comment by bnoordhuis** [comment]
CI: https://ci.nodejs.org/job/libuv-test-commit/16/

File: test/test-tcp-close-accept.c (hunk 1 above)
Comment (santigimeno) [inline test/test-tcp-close-accept.c]: Thanks for the suggestion and explanation, I'll update the PR accordingly. Test still failing, fix coming

**Comment by santigimeno** [comment]
PR updated. I *think* it should work now (can't reproduce the error locally). Thanks!

**Comment by cjihrig** [comment]
CI: https://ci.nodejs.org/job/libuv-test-commit/17/

File: test/test-tcp-close-accept.c [hunk 2]
```diff
@@ -58,8 +59,8 @@ static void connect_cb(uv_connect_t* req, int status) {
   if (req == &tcp_check_req) {
     ASSERT(status != 0);
 
-    /* Close check and incoming[0], time to finish test */
-    uv_close((uv_handle_t*) &tcp_incoming[0], close_cb);
+    /* Close check and pending incoming connection, time to finish test */
```
Comment (saghul) [inline test/test-tcp-close-accept.c]: can you add some articles here? :-)

**Comment by saghul** [comment]
LGTM with a small comment.

**Comment by santigimeno** [comment]
Comment updated. Thanks

**Comment by saghul** [comment]
LGTM

**Comment by saghul** [comment]
Sorry for the delay! Landed in 51d660d, thanks Santi!

**Pull request closed without merging**